specifyweb/specify/build_models.py

```diff
@@ -63,9 +64,18 @@ def save(self, *args, **kwargs):
             return super(model, self).save(*args, **kwargs)
         except AbortSave:
             return
+        
+    def avant_delete(self):
```

Comment:
also, I'm not sure what does the name of this function mean
is this like `pre_delete`? or `pre_pre_delete`? 😊

Reply:
lol, ya, I thought `pre_pre_delete` sounded dumb, so I instead used the french word "avant" meaning before, as to distinct itself from `pre_delete`.  I wrote a pretty lengthy description just to clarify its use.  I'll take any ideas for another name.

It is unfortunate that it needed to be done this way.  The problem was that Django enforces database constraints prior to the `pre_delete` signal.  So, Django returned errors before the `pre_delete` function could run.  Adding this new handler function was the simplest and most canonical way to incorporate it into the existing business rules code.  At least with this solution, we can easily extend the functionality to any other time this problem comes up.

https://github.com/specify/specify7/blob/3a548c3125d4917d26958256d89b340d752bbc67/specifyweb/specify/build_models.py#L68-L73

Reply:
decided on going with `pre_constraints_delete` instead